x/crosschain/keeper/cctx_orchestrator_validate_inbound.go

```diff
@@ -0,0 +1,50 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+	observertypes "github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+// ValidateInbound is the only entry-point to create new CCTX (eg. when observers voting is done or new inbound event is detected).
+// It creates new CCTX object and calls InitiateOutbound method.
+func (k Keeper) ValidateInbound(
```

Comment:
Do you think Validate is still an accurate name for this function? 
IMO, it does more than just validate .

What do you think about breaking this into validate and process steps? 

This part is more about processing that validating 
https://github.com/zeta-chain/zeta-node/blob/2c2d7e5462f004d828371d3307959069fa1770f7/x/crosschain/keeper/cctx_orchestrator_validate_inbound.go#L35-L48

Reply:
okay , but I do strongly feel that function names like validate should ideally be functions that do not do state transitions , as that's the design we have been trying to implement across the codebase .

Please feel free to resolve this , but we should discuss this soon, before implementing other logic so that the refactor doesn't become more difficult later